World/Magic/Spell.cs

```diff
@@ -24,15 +24,36 @@ public void GetSpoken(Player player)
             if (!result.Success)
             {
                 player.Echo("Your spell fizzles!");
+                return;
             }
-            else if (result.Action != null)
+            if (result.Action == null)
             {
-                result.Action();
+                player.Echo("Your spell returned: {result.Value}");
+                return;
             }
-            else
+            // assumption: only have cost when have an action
+            var resources = result.PossibleSources(player);
+            var completelyConsumed = this.consume(result.Cost, resources);
+            if (!completelyConsumed)
             {
-                player.Echo("Your spell returned: {result.Value}");
+                player.Echo("Your spell fizzles!");
+                return;
+            }
+            result.Action();
+        }
+
+        public bool consume(long cost, IPowerSource[] resources)
+        {
+            foreach (IPowerSource ps in resources)
+            {
+                var actualConsumption = ps.consumeTotal(cost);
+                cost -= actualConsumption;
+                if (cost == 0)
```

Comment:
ik zou er minder dan of gelijk 0 van maken, anders kan je met een klein foutje ergens in een powersource zorgen dat andere sources ook nog gaan drawen